src/analytics/api/utils/pollutants/report.py

```
@@ -29,36 +28,32 @@ def convert_utc_to_local(timestamps, site_latitude, site_longitude):
+        list: Site IDs belonging to the grid; empty on query failure.
     """
-    grid_params = {
-        "token": settings.airqo_api_token.get_secret_value(),
-        "startTime": start_time.isoformat() + "Z",
-        "endTime": end_time.isoformat() + "Z",
-        "recent": "no",
-    }
-
-    grid_url = f"{settings.grid_url}{grid_id}"
+    grids_sites_table = Utils.table_name(settings.bigquery_grids_sites)
+    query = f"SELECT DISTINCT site_id FROM {grids_sites_table} WHERE grid_id = @grid_id"
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)